learn2learn/algorithms/lightning/lightning_episodic_module.py

```diff
@@ -92,26 +102,34 @@ def training_step(self, batch, batch_idx):
         return train_loss
 
     def validation_step(self, batch, batch_idx):
-        valid_loss, valid_accuracy = self.meta_learn(
-            batch, batch_idx, self.test_ways, self.test_shots, self.test_queries
-        )
-        self.log(
-            "valid_loss",
-            valid_loss.item(),
-            on_step=False,
-            on_epoch=True,
-            prog_bar=False,
-            logger=True,
-        )
-        self.log(
-            "valid_accuracy",
-            valid_accuracy.item(),
-            on_step=False,
-            on_epoch=True,
-            prog_bar=True,
-            logger=True,
-        )
-        return valid_loss.item()
+        if self.should_fit_on_validate:
+            # used for the algorithm to store the supports data
+            self.cache_on_validate_step(batch, batch_idx)
+        else:
+            valid_loss, valid_accuracy = self.meta_learn(
+                batch, batch_idx, self.test_ways, self.test_shots, self.test_queries
+            )
+            self.log(
+                "valid_loss",
+                valid_loss.item(),
+                on_step=False,
+                on_epoch=True,
+                prog_bar=False,
+                logger=True,
+            )
+            self.log(
+                "valid_accuracy",
+                valid_accuracy.item(),
+                on_step=False,
+                on_epoch=True,
+                prog_bar=True,
+                logger=True,
+            )
+            return valid_loss.item()
```

Comment:
Do we still reach this branch when validating during training? (I'm not sure when `TrainerFn.VALIDATING` is `True` or `False`.)

Reply:
This part of the code is run only when the users use `trainer.validate`. This is a hack and need to find a better solution.
Another option would be to provide 2 predict dataloaders, but there would be some hacks.

Reply:
Removed ! Would be good to brainstorm together on the best way to support this.